tools/clang/test/HLSLFileCheck/hlsl/workgraph/record_type_metadata.ll

```diff
@@ -0,0 +1,46 @@
+; RUN: %dxilver 1.8 | %D3DReflect %s | FileCheck %s -check-prefixes=RDAT
+
+; Make sure NodeRecordType metdata loading is robust to tag,value list ordering.
+
+; RDAT: FunctionTable[{{.*}}] = {
+; RDAT-LABEL: UnmangledName: "node01"
+; RDAT: Inputs: <11:RecordArrayRef<IONode>[1]>  = {
+; RDAT: AttribKind: RecordSizeInBytes
+; RDAT-NEXT: RecordSizeInBytes: 12
+; RDAT: AttribKind: RecordDispatchGrid
+; RDAT-NEXT: RecordDispatchGrid: <RecordDispatchGrid>
+; RDAT-NEXT: ByteOffset: 0
+; RDAT-NEXT: ComponentNumAndType: 23
```

Comment:
Where does this 23 come from?

Reply:
This is a combination of `NumComponents` (`3`) and `DXIL::ComponentType::U32` (`5`) `<< 2`:
`0:2 = NumComponents (0-3)`
`3:15 = hlsl::DXIL::ComponentType enum`

https://github.com/microsoft/DirectXShaderCompiler/blob/2872d2d0e7eeec9c4caec91ff3858f4f54f31f7b/include/dxc/DxilContainer/RDAT_LibraryTypes.inl#L322-L323